src/metabase/driver/mysql/actions.clj

```diff
@@ -52,66 +40,82 @@
       [[] nil nil]
       (jdbc/reducible-query jdbc-spec sql-args {:identifers identity, :transaction? false})))))
 
-(defn- maybe-parse-unique-constraint-error [database error-message]
-  (let [[match table constraint]
-        (re-find #"Duplicate entry '.+' for key '(.+)\.(.+)'" error-message)
-        constraint (remove-backticks constraint)
-        table (remove-backticks table)]
-    (when match
-      (some->> (constraint->column-names database table constraint)
-               (mapv (fn [col]
-                       {:message    (tru "violates unique constraint {0}" constraint)
-                        :table      table
-                        :constraint constraint
-                        :column     col}))))))
+(defn- remove-backticks [id]
+  (when id
+    (-> id
+        (str/replace "``" "`")
+        (str/replace #"^`?(.+?)`?$" "$1"))))
+
+(defmethod sql-jdbc.actions/maybe-parse-sql-error [:mysql actions.error/violate-not-null-constraint]
+  [_driver error-type _database _action-type error-message]
+  (or
+   (when-let [[_ column]
+              (re-find #"Column '(.+)' cannot be null" error-message)]
+     {:type    error-type
+      :message (tru "{0} must have values." (str/capitalize column))
+      :errors  {column (tru "You must provide a value.")}})
+   (when-let [[_ column]
+              (re-find #"Field '(.+)' doesn't have a default value" error-message)]
+     {:type    error-type
+      :message (tru "{0} must have values." (str/capitalize column))
+      :errors  {column (tru "You must provide a value.")}})))
+
+(defmethod sql-jdbc.actions/maybe-parse-sql-error [:mysql actions.error/violate-unique-constraint]
+  [_driver error-type _database _action-type error-message]
+  (when-let [[_match fk]
+             (re-find #"Duplicate entry '.+' for key '(.+)'" error-message)]
+    (let [column (last (str/split fk #"\."))]
+      {:type    error-type
+       :message (tru "{0} already exist." (str/capitalize column))
```

Comment:
There's only one column, so this needs to be
```suggestion
       :message (tru "{0} already exists." (str/capitalize column))
```

Reply:
good catch. Thanks